python/node_classification.py

```diff
@@ -0,0 +1,421 @@
+            loss=logged_data[k]["loss"],
+            train_log=logged_data[k]["train"],
+            val_log=logged_data[k]["val"],
+        )
+        for k in range(len(logged_data))
+    ]
+
+
+##############################
+# get training data
+##############################
+
+
+@mgp.read_proc
+def get_training_data() -> mgp.Record(
+    epoch=int, loss=float, train_log=mgp.Any, val_log=mgp.Any
+):
+    """This function is used so user can see what is logged data from training.
+
+
+    Returns:
+        mgp.Record(
+            epoch (int): epoch number of record of logged data row
+            loss (float): loss in logged data row
+            train_log (mgp.Any): training parameters of record of logged data row
+            val_log (mgp.Any): validation parameters of record of logged data row
+            ): record to return
+
+
+    """
+
+    return [
+        mgp.Record(
+            epoch=logged_data[k]["epoch"],
+            loss=logged_data[k]["loss"],
+            train_log=logged_data[k]["train"],
+            val_log=logged_data[k]["val"],
+        )
+        for k in range(len(logged_data))
+    ]
+
+
+##############################
+# model loading and saving, predict
+##############################
+
+
+@mgp.read_proc
+def save_model() -> mgp.Record(path=str):
+    """This function saves model to previously defined path_to_model.
+
+    Returns:
+        mgp.Record(path (str): path to model): return record
+    """
+
+    if Modelling.model == None:
+        raise AssertionError("model is not loaded")
+    torch.save(Modelling.model.state_dict(), global_params[OtherParams.PATH_TO_MODEL])
+    return mgp.Record(path=global_params[OtherParams.PATH_TO_MODEL])
+
+
+@mgp.read_proc
+def load_model() -> mgp.Record(path=str):
+    """This function loads model to previously defined path_to_model.
+
+    Returns:
+        mgp.Record(path (str): path to model): return record
+    """
+    global model
+
+    if not os.path.exists(os.path.abspath(global_params[OtherParams.PATH_TO_MODEL])):
+        raise Exception("file not found on path")
+
+    Modelling.model.load_state_dict(
+        torch.load(global_params[OtherParams.PATH_TO_MODEL])
+    )
+    return mgp.Record(path=global_params[OtherParams.PATH_TO_MODEL])
+
+
+@mgp.read_proc
+def predict() -> mgp.Record(dict=mgp.Map):
+    """This function predicts metrics on all nodes. It is suggested that user previously
+    loads unseen test data to predict on it.
+
+    Returns:
+        mgp.Record(mgp.Map: dictionary of all metrics): record to return
+    """
+
+    dict = metrics(
+        Modelling.data.train_mask + Modelling.data.val_mask,
+        Modelling.model,
+        Modelling.data,
+        global_params[DataParams.METRICS],
+    )
+
+    return mgp.Record(dict=dict)
+
+
+@mgp.read_proc
+def reset():
+    if "global_params" in globals().keys():
+        del globals()["global_params"]
+
+    if "logged_data" in globals().keys():
+        del globals()["logged_data"]
```

Comment:
The `globals()` method returns a dictionary with all the global variables and symbols for the current program. I used to writing backend in PHP, it has similar syntax with $GLOBALS. Thanks!

Reply:
Not sure how good of practice is to reference the `globals` like that. Since no other module does it, maybe it would be better to opt for a direct referencing of the variable.